sdk/attestation/Azure.Security.Attestation/src/Models/AttestationTokenSigningKey.cs

```diff
@@ -62,17 +66,26 @@ public AttestationTokenSigningKey(X509Certificate2 certificate)
 
         private static void VerifySignerMatchesCertificate(AsymmetricAlgorithm signer, X509Certificate2 certificate)
         {
-            if (!signer.KeyExchangeAlgorithm.StartsWith(certificate.PublicKey.Key.KeyExchangeAlgorithm, System.StringComparison.Ordinal))
+#if NET6_0_OR_GREATER
+            AsymmetricAlgorithm publicKey = (AsymmetricAlgorithm)certificate.GetRSAPublicKey() ?? certificate.GetDSAPublicKey();
+#else
+            AsymmetricAlgorithm publicKey = certificate.PublicKey.Key;
```

Comment:
See: [SYSLIB0027: PublicKey.Key is obsolete](https://learn.microsoft.com/en-us/dotnet/fundamentals/syslib-diagnostics/syslib0027)